### DISCUSSION TITLE
Regarding the conditions for downloading the engine folder and below with prisma generate

### DISCUSSION THREAD

**Discussion by nrikiji** [prisma/prisma#23101] [Q&A] [answered]
### Question

After `prisma generate` in one environment, copy `node_modules` and `schema.prisma` to another environment.
If you try to `prisma generate` again in this new environment (same system configuration), the `engine` folder and below will be downloaded again.

Are there any files other than `node_modules` that need to be copied?

**Comment by ludralph** [comment, marked as answer]
Hi @nrikiji 👋 

Thank you for raising this question.

The `prisma generate` command primarily interacts with the `node_modules` directory and the `schema.prisma` file. When you run `prisma generate`, it downloads the necessary [engine files](https://www.prisma.io/docs/orm/more/under-the-hood/engines#using-custom-engine-libraries-or-binaries) into the `node_modules/@prisma/engines` folder and also copies the query engine to the generated Prisma Client.

However, it's important to note that Prisma also creates a default [.env file](https://www.prisma.io/docs/orm/more/development-environment/environment-variables/env-files) at your project's root, which is used to configure your database connection URL. If you're moving your project to a new environment, you might also need to copy this .env file, especially if it contains environment-specific variables.

In summary, you should ensure that you've copied the `node_modules` directory, the `schema.prisma` file, and the `.env` file when moving your Prisma project to a new environment.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nrikiji** [reply]
While it hasn't resolved the issue in my project, I understand the overview. There seems to be an issue in the copying process. I'll consider this discussion closed for now. Thank you.
